### PR TITLE
tests/e2e: Add support to create cluster with operator

### PR DESCRIPTION
**Pull request by kfirtoledo**
Add options to create Clustrlink using the operator in the e2e tests.

### CODE REVIEW

File: tests/e2e/k8s/util/kind.go
Comment (orozery) [inline tests/e2e/k8s/util/kind.go:157]:
```suggestion
	// Add clusterlink CRDs to scheme.
```

Also, my personal reference is to rename an import (i.e. `v1alpha1` to `clusterlink`) only when package names collide. But your call.

File: tests/e2e/k8s/util/kind.go
Comment (orozery) [inline tests/e2e/k8s/util/kind.go:373]: I think folder is a Windows term. Perhaps rename to Directory?

File: tests/e2e/k8s/util/kind.go
Comment (orozery) [inline tests/e2e/k8s/util/kind.go:376]:
```suggestion
	return decoder.DecodeEachFile(context.Background(), fs, pattern,
```

File: tests/e2e/k8s/util/kind.go
Comment (orozery) [inline tests/e2e/k8s/util/kind.go:375]: I would remove this lines, and embed the values directly to the function call below.

File: tests/e2e/k8s/util/kind.go
Comment (orozery) [inline tests/e2e/k8s/util/kind.go:501]: I'm trying to keep this file to be general kind-utils, and not clusterlink specific.
I would remove those functions, and inline their code in the caller.

File: tests/e2e/k8s/util/fabric.go
Comment (orozery) [inline tests/e2e/k8s/util/fabric.go:166]: Just noticed this const. NameSpace -> Namespace

**Review by orozery — CHANGES_REQUESTED** [review]

File: tests/e2e/k8s/util/kind.go
Comment (elevran) [reply tests/e2e/k8s/util/kind.go:373]: Or `path`.
Would it also make sense to validate that the parameter is indeed a directory and not a file or let `DirFS` fail?

**Comment by kfirtoledo** [comment]
@orozery PTAL (After the approval I will sync it).

File: pkg/operator/controller/instance_controller.go
Comment (orozery) [inline pkg/operator/controller/instance_controller.go:50]: Why is the default operator namespace different than the instance namespace?

File: tests/e2e/k8s/suite.go
Comment (orozery) [inline tests/e2e/k8s/suite.go:101]:
```suggestion

		// create the project operator.
```

File: tests/e2e/k8s/util/fabric.go
Comment (orozery) [inline tests/e2e/k8s/util/fabric.go:32]: Move these to the upper import group.

File: tests/e2e/k8s/util/fabric.go
Comment (orozery) [inline tests/e2e/k8s/util/fabric.go:45]:
```suggestion
	// DeployWithOperator deploys clusterlink using an operator.
```

File: tests/e2e/k8s/util/kind.go
Comment (orozery) [inline tests/e2e/k8s/util/kind.go:26]: move this import to its own import group downwards.

**Review by orozery — CHANGES_REQUESTED** [review]

File: pkg/operator/controller/instance_controller.go
Comment (kfirtoledo) [reply pkg/operator/controller/instance_controller.go:50]: Usually, the operator runs in a different namespace with more privileges (like in openshift there is an openshift-operator namespace)

**Review by orozery — APPROVED** [review]

**Pull request merged**
